### PR TITLE
HIVE-25266: Fix TestWarehouseExternalDir

### PR DESCRIPTION
**Pull request by rkirtir**
### What changes were proposed in this pull request?
Fix TestWarehouseExternalDir

### Why are the changes needed?
 TestWarehouseExternalDir was flaky

### Does this PR introduce _any_ user-facing change?
NO

### How was this patch tested?
http://ci.hive.apache.org/job/hive-flaky-check/679/
http://ci.hive.apache.org/job/hive-flaky-check/680/
http://ci.hive.apache.org/job/hive-flaky-check/681/

### CODE REVIEW

**Comment by rkirtir** [comment]
This test was tested via flaky test 
http://ci.hive.apache.org/job/hive-flaky-check/681/

**Comment by aturoczy** [comment]
LGTM

File: itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/TestWarehouseExternalDir.java [hunk 1]
```diff
@@ -44,114 +44,94 @@
...
+  private  static final Logger LOG = LoggerFactory.getLogger(TestWarehouseExternalDir.class);
+
+  private static MiniHS2 miniHS2;
+  private static Hive db;
 
-  static MiniHS2 miniHS2;
-  static Hive db;
-  static Connection conn;
+  private static HiveConf conf;
+  private  static Connection conn;
```
Comment (veghlaci05) [inline itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/TestWarehouseExternalDir.java:59]: I would remove this field. 

1. `beforeTest()` creates a connection (line 88)
2. This is **_not closed_** but reassigned with a new connection in setUp() (line 119)
3. `tearDown()` closes the connection after every test (but there's only one test)

Even despite the lot of maintenance code this results in an unclosed connection. I would simply create and properly close local connection variables both in `beforeTest()` and `testExternalDefaultPaths()`

**Review by veghlaci05 — CHANGES_REQUESTED** [review]

File: itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/TestWarehouseExternalDir.java [hunk 2]
```diff
@@ -174,19 +155,19 @@ public void testManagedPaths() throws Exception {
   @Test
   public void testExternalDefaultPaths() throws Exception {
     try (Statement stmt = conn.createStatement()) {
-      stmt.execute("create external table default.twed_ext1(c1 string)");
-      Table tab = db.getTable("default", "twed_ext1");
+      stmt.execute("create external table IF NOT EXISTS default.twed_ext1(c1 string)");
```
[the remark this answers is not in the document]
Comment (rkirtir) [reply itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/TestWarehouseExternalDir.java:156]: fixed

File: itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/TestWarehouseExternalDir.java (hunk 1 above)
Comment (rkirtir) [reply itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/TestWarehouseExternalDir.java:59]: Agree.
Removed unnecessary conn creation and closed connection in afterTest() class

**Comment by aturoczy** [comment]
I think it needs to be retrigger. Seems like this test is flaky:

java.util.NoSuchElementException
	at java.util.LinkedList.getLast(LinkedList.java:257)
	at org.apache.hadoop.hive.ql.parse.TestReplicationScenariosAcidTables.testSizeOfDatabaseReplication(TestReplicationScenariosAcidTables.java:3980)
	at org.apache.hadoop.hive.ql.parse.TestReplicationScenariosAcidTables.testSizeOfDatabaseReplicationViaDistCp(TestReplicationScenariosAcidTables.java:3911)

**Review by veghlaci05 — APPROVED** [review]
LGTM

**Pull request merged**